src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -462,14 +472,20 @@ def execute(self):
             classpath_product.remove_for_target(cc.target, [(conf, cc.classes_dir)])
             classpath_product.add_for_target(cc.target, [(conf, cc.jar_file)])
 
+      # The runtime classpath will always be a superset of the export_dep_as_jar_classpath
+      # so however we compute runtime_classpath it will contain all the nescessary jars for
+      # the export_dep_as_jar_classpath.export_dep_as_jar_classpath.
+      if self.context.products.is_required_data('export_dep_as_jar_classpath'):
+        self.context.products.get_data('export_dep_as_jar_classpath', classpath_product.copy)
```

Comment:
It's fine for now, but `get_data` isn't guaranteed to have the sideeffect of calling the initializer: if you ever installed another task to compute additional entries for `export_dep_as_jar_classpath` and it ran first, this would break, because `copy` would not be called.

Reply:
Good to know for the future, thanks!